app/controllers/users_controller.rb

```diff
@@ -34,6 +34,16 @@ def add_language
     end
   end
 
+  def remove_language
+    authorize @language
+    current_user.languages.delete @language
+    if current_user.save
+      redirect_to edit_users_path, notice: "#{@language.name} was removed from your languages list."
+    else
+      redirect_to edit_users_path, notice: "Error unable to remove #{@language.name} from your languages list!"
+    end
+  end
```

Comment:
It might be water under the bridge at this point, especially since the `add_language` behavior was already moved to this controller, but we do now have the `UserLanguage` model. Does it makes sense to add a `UserLanguagesController` with say a `create` action and a `destroy` action to create/destroy this association? It would be a bit more RESTful and the logic wouldn't really have to change just the location of the code.

Reply:
Seems reasonable : )